cpp/models/ode_secirvvs/parameters.h

```diff
@@ -644,82 +648,182 @@ class Parameters : public ParametersBase
                             this->get<RecoveredPerInfectedNoSymptoms>()[i], 0);
                 this->get<RecoveredPerInfectedNoSymptoms>()[i] = 0;
+                corrected                                      = 1;
             }
 
             if (this->get<RiskOfInfectionFromSymptomatic>()[i] < 0.0 ||
                 this->get<RiskOfInfectionFromSymptomatic>()[i] > 1.0) {
                 log_warning("Constraint check: Parameter RiskOfInfectionFromSymptomatic changed from {:0.4f} to {:d}",
                             this->get<RiskOfInfectionFromSymptomatic>()[i], 0);
                 this->get<RiskOfInfectionFromSymptomatic>()[i] = 0;
+                corrected                                      = 1;
             }
 
             if (this->get<SeverePerInfectedSymptoms>()[i] < 0.0 || this->get<SeverePerInfectedSymptoms>()[i] > 1.0) {
                 log_warning("Constraint check: Parameter SeverePerInfectedSymptoms changed from {:0.4f} to {:d}",
                             this->get<SeverePerInfectedSymptoms>()[i], 0);
                 this->get<SeverePerInfectedSymptoms>()[i] = 0;
+                corrected                                 = 1;
             }
 
             if (this->get<CriticalPerSevere>()[i] < 0.0 || this->get<CriticalPerSevere>()[i] > 1.0) {
                 log_warning("Constraint check: Parameter CriticalPerSevere changed from {:0.4f} to {:d}",
                             this->get<CriticalPerSevere>()[i], 0);
                 this->get<CriticalPerSevere>()[i] = 0;
+                corrected                         = 1;
             }
 
             if (this->get<DeathsPerCritical>()[i] < 0.0 || this->get<DeathsPerCritical>()[i] > 1.0) {
                 log_warning("Constraint check: Parameter DeathsPerCritical changed from {:0.4f} to {:d}",
                             this->get<DeathsPerCritical>()[i], 0);
                 this->get<DeathsPerCritical>()[i] = 0;
+                corrected                         = 1;
+            }
+
+            if (this->get<DaysUntilEffectivePartialImmunity>()[i] < 0.0) {
+                log_warning("Constraint check: Parameter DeathsPerCritical changed from {:0.4f} to {:d}",
+                            this->get<DaysUntilEffectivePartialImmunity>()[i], 0);
+                this->get<DaysUntilEffectivePartialImmunity>()[i] = 0;
+                corrected                                         = 1;
+            }
+            if (this->get<DaysUntilEffectiveImprovedImmunity>()[i] < 0.0) {
+                log_warning(
+                    "Constraint check: Parameter DaysUntilEffectiveImprovedImmunity changed from {:0.4f} to {:d}",
+                    this->get<DaysUntilEffectiveImprovedImmunity>()[i], 0);
+                this->get<DaysUntilEffectiveImprovedImmunity>()[i] = 0;
+                corrected                                          = 1;
+            }
+
+            if (this->get<ReducExposedPartialImmunity>()[i] <= 0.0 ||
```

Comment:
Shouldn't this be a <0.0 instead of <=? Because when you correct it, it is set to 0.

Reply:
Thanks for the comment. In model.h we divide by this parameter. So of course it should not be set to 0 :)